File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        unordered_set<int> st;
        int min = INT_MAX;
        int ans = 1;
        for(int num : nums){
            st.insert(num);
        }
        for(auto it : st) {
            int cnt = 0;
            if(st.find(it - 1) == st.end()){
                min = it;
                cnt = 1;
                while(st.find(min+1) != st.end()){
                    cnt++;
                    min++;
                }
                ans = max(ans, cnt);
            }
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int n = nums.length;
        HashSet<Integer> set = new HashSet<>();
        for(int num : nums){
           set.add(num);
        }
        int longestStreak = 0;
        for(int num : set){
            if(!set.contains(num - 1)){
                // int num = nums[i];
                int currStreak = 1;
                while(set.contains(num+1)){
                    num++;
                    currStreak++;
                }
                longestStreak = Math.max(longestStreak, currStreak);
            }
        }
        return longestStreak;
    }
}